Source/RapyutaSimulationPlugins/Private/Tools/SimulationState.cpp

```diff
@@ -72,141 +72,92 @@ void ASimulationState::AddSpawnableEntities(TMap<FString, TSubclassOf<AActor>> I
     }
 }
 
-void ASimulationState::GetEntityStateSrv(UROS2GenericSrv* Service)
+template<typename T>
+bool ASimulationState::CheckEntity(TMap<FString, T> InEntities, const FString & InEntityName, const bool AllowEmpty)

```

Comment:
Might the params be `const TMap<FString, T>& InEntities` & `bAllowEmpty` (`b` prefix also for all below)?